### PR TITLE
Adding a Vagrantfile for spinning up a development VM

### PR DESCRIPTION
**Pull request by IdanZimi**
Vagrant is a tool for using virtual machines to share development environments. Learn more about it at:
https://www.vagrantup.com/

Also making changes to the `.gitignore` file to ignore any data that Vagrant might create.

### CODE REVIEW

**Comment by Danielsio** [comment]
I don't see any issues. Great job.

**Comment by amitKaplansky** [comment]
looks good to me.

File: .gitignore [hunk 1]
```diff
@@ -127,3 +127,6 @@ dmypy.json
 
 # Pyre type checker
 .pyre/
+
+# Vagrant
+.vagrant
```
Comment (yftacherzog) [inline .gitignore:132]: add blank link at end of file.

**Review by yftacherzog — CHANGES_REQUESTED** [review]

File: Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  # For a complete reference, please see the online documentation at
+  # https://docs.vagrantup.com.
+
+  # Every Vagrant development environment requires a box. You can search for
+  # boxes at https://vagrantcloud.com/search.
+  config.vm.box = "fedora/34-could-base"
+  config.vm.synced_folder ".", "/vagrant", type: "virtualbox"	
+  config.vm.provider "virtualbox" do |vb|
+  vb.memory = "1024"
+  vb.cpus = "2"
```
[the remark this answers is not in the document]
Comment (IdanZimi) [reply Vagrantfile]: fixed indentation. is it ok?

File: .gitignore (hunk 1 above)
Comment (IdanZimi) [reply .gitignore]: Added blank line at end of file. waiting for a review.

**Comment by OhadSeg** [comment]
Seems to me that you made the necessary changes.

File: Vagrantfile (hunk 2 above)
[the remark this answers is not in the document]
Comment (yftacherzog) [reply Vagrantfile]: Can you test locally that you are able to start the vm? Or at least fail while running setup.sh and not earlier?

**Comment by Danielsio** [comment]
I think it's good to go

**Review by yftacherzog — APPROVED** [review]

**Pull request merged**